moveit_core/robot_state/include/moveit/robot_state/robot_state.h

```diff
@@ -1198,6 +1249,47 @@ as the new values that correspond to the group */
   static double testAbsoluteJointSpaceJump(const JointModelGroup* group, std::vector<RobotStatePtr>& traj,
                                            double revolute_jump_threshold, double prismatic_jump_threshold);
 
+  /** \brief Tests for large Cartesian space jumps of a trajectory at a specific link.
+
+     Takes the midpoint between points in the trajectory and solves the FK. If the pose at the midpoint is further than
+     the max_step from either the preceding point or the following point, then the returned path is truncated up to
+     just before the jump
+
+     @param group The joint model group of the robot state.
+     @param link The link used for testing
+     @param traj The trajectory that should be tested.
+     @param max_step A struct containing the maximum translation and rotation between waypoints
+     @return The fraction of the trajectory that passed.
+  */
+  static double testCartesianSpaceJump(const JointModelGroup* group, const LinkModel* link,
```

Comment:
the function name 'test' bothers me since its not a *unit test*, i think we should reserve that keyword for actual tests. "check" would be much better. note that the convention of calling it a "test" was introduced by @hartmanndennis [only last September](https://github.com/ros-planning/moveit/pull/576/files) (i approved the PR)

Reply:
me too. I like trim the best but would also be ok with check.